git-publish

```diff
@@ -136,6 +147,7 @@ def git_get_current_branch():
     if os.path.exists(rebase_dir):
         branch_path = os.path.join(rebase_dir, 'head-name')
         prefix = 'refs/heads/'
+        # The result branch name is valied encoded
         branch = open(branch_path).read().strip()
```

Comment:
I think this is incorrect because the `git-log(1)` documentation says:
```
Path names are encoded in UTF-8 normalization form C. This applies to tree objects, the index file, ref names, as well as path names in command line arguments, environment variables and config files
```

`open(branch_path).read()` reads the file in text mode using Python's encoding. This file contains a ref name and the git documentation says ref names are encoded using UTF-8.

So I think this should be:
```python
branch = open(branch_path, encoding=GIT_ENCODING).read().strip()
```